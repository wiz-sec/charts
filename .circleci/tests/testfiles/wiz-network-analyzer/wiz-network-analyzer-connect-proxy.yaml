wizApiToken:
  clientId: "client-id"
  clientToken: "client-secret"

httpProxyConfiguration:
  enabled: true
  httpsProxy: http://mitmweb:8080

caCertificate:
  enabled: true
  certificate: |
    -----BEGIN CERTIFICATE-----
    abcd1234
    -----END CERTIFICATE-----
