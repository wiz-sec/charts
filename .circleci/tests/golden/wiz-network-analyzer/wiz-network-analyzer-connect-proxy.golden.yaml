---
# Source: wiz-network-analyzer/templates/service-account-cluster-reader.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: wiz-network-analyzer-cluster-reader
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-network-analyzer
    app.kubernetes.io/instance: release-test
    app.kubernetes.io/version: "0.1"
    app.kubernetes.io/managed-by: Helm
rules:
  - apiGroups:
    - '*'
    resources:
    - '*'
    verbs:
    - get
    - list
    - watch
---
# Source: wiz-network-analyzer/templates/service-account-cluster-reader.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: wiz-network-analyzer-cluster-reader
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-network-analyzer
    app.kubernetes.io/instance: release-test
    app.kubernetes.io/version: "0.1"
    app.kubernetes.io/managed-by: Helm
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name:  wiz-network-analyzer-cluster-reader
subjects:
- kind: ServiceAccount
  name: wiz-network-analyzer-cluster-reader
  namespace: "release-helm-namespace"
---
# Source: wiz-network-analyzer/templates/job-network-analyzer.yaml
apiVersion: batch/v1
kind: Job
metadata:
  name: wiz-network-analyzer
  namespace: "release-helm-namespace"
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-network-analyzer
    app.kubernetes.io/instance: release-test
    app.kubernetes.io/version: "0.1"
    app.kubernetes.io/managed-by: Helm
  annotations:
    rollme.wizApiTokenHash: 6d187a29d523282461d9087dc16d037fa8ffc8e309c077db44a1c4196e4583dd
    rollme.proxyHash: cc1bd4c7aa97d4eaa12689b782068dbfb3854cb0aa78d1ecfa6080e48cd22d33

spec:
  ttlSecondsAfterFinished: 300
  manualSelector: true
  selector:
    matchLabels:
      app.kubernetes.io/name: wiz-network-analyzer
      app.kubernetes.io/instance: release-test
  backoffLimit: 1
  template:
    metadata:
      labels:
        helm.sh/chart: "GOLDEN_STATIC_VALUE"
        app.kubernetes.io/name: wiz-network-analyzer
        app.kubernetes.io/instance: release-test
        app.kubernetes.io/version: "0.1"
        app.kubernetes.io/managed-by: Helm
    spec:
      serviceAccountName: wiz-network-analyzer-cluster-reader
      restartPolicy: "Never"
      securityContext:
        runAsNonRoot: true
        runAsUser: 1000
      volumes:
      - name: additional-certs
        secret:
          secretName: release-test-na-ca
          items:
            - key: ca.crt
              path: ca-certificates.crt
          defaultMode: 420
      
      
      containers:
        - name: wiz-network-analyzer
          securityContext:
            allowPrivilegeEscalation: false
            runAsNonRoot: true
            runAsUser: 1000
          image: wiziopublic.azurecr.io/wiz-app/wiz-network-analyzer:0.1
          imagePullPolicy: Always
          command:
            - "wiz-network-analyzer"
          args:
            
            - analyze
          env:
          - name: LOG_LEVEL
            value: info
          - name: WIZ_CLIENT_ID
            valueFrom:
              secretKeyRef:
                name: release-test-na-api-token
                key: clientId
                optional: false
          - name: WIZ_CLIENT_TOKEN
            valueFrom:
              secretKeyRef:
                name: release-test-na-api-token
                key: clientToken
                optional: false
          - name: WIZ_CLIENT_SECRET
            valueFrom:
              secretKeyRef:
                name: release-test-na-api-token
                key: clientToken
                optional: false
          - name: WIZ_ENV
            value: 
          - name: HTTPS_PROXY
            valueFrom:
              secretKeyRef:
                name: release-test-na-proxy-configuration
                key: httpsProxy
                optional: false
          - name: NO_PROXY
            valueFrom:
              secretKeyRef:
                name: release-test-na-proxy-configuration
                key: noProxyAddress
                optional: false
          - name: SSL_CERT_DIR
            value: /usr/local/share/ca-certificates/:/certificates/:/etc/ssl/certs
          resources:
            null
          volumeMounts:
          - name: additional-certs
            mountPath: /certificates
          - name: additional-certs
            mountPath: /etc/ssl/certs
---
# Source: wiz-network-analyzer/templates/service-account-cluster-reader.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: wiz-network-analyzer-cluster-reader
  namespace: "release-helm-namespace"
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-network-analyzer
    app.kubernetes.io/instance: release-test
    app.kubernetes.io/version: "0.1"
    app.kubernetes.io/managed-by: Helm
  annotations:
    "helm.sh/hook": pre-install,pre-upgrade
    "helm.sh/hook-delete-policy": before-hook-creation
---
# Source: wiz-network-analyzer/templates/secret-ca.yaml
apiVersion: v1
kind: Secret
type: Opaque
metadata:
  name: release-test-na-ca
  namespace: "release-helm-namespace"
  annotations:
    "helm.sh/hook": pre-install,pre-upgrade
    "helm.sh/hook-weight": "-1"
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-network-analyzer
    app.kubernetes.io/instance: release-test
    app.kubernetes.io/version: "0.1"
    app.kubernetes.io/managed-by: Helm
stringData:
  ca.crt: "-----BEGIN CERTIFICATE-----\nabcd1234\n-----END CERTIFICATE-----\n"
---
# Source: wiz-network-analyzer/templates/secret-proxy.yaml
apiVersion: v1
kind: Secret
type: Opaque
metadata:
  name: release-test-na-proxy-configuration
  namespace: "release-helm-namespace"
  annotations:
    "helm.sh/hook": pre-install,pre-upgrade
    "helm.sh/hook-weight": "-1"
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-network-analyzer
    app.kubernetes.io/instance: release-test
    app.kubernetes.io/version: "0.1"
    app.kubernetes.io/managed-by: Helm
stringData:
  httpsProxy: "http://mitmweb:8080"
  noProxyAddress: "kubernetes,kubernetes.default,kubernetes.default.svc,kubernetes.default.svc.cluster.local"
---
# Source: wiz-network-analyzer/templates/secrets-wiz-api-token.yaml
apiVersion: v1
kind: Secret
metadata:
  name: release-test-na-api-token
  namespace: "release-helm-namespace"
  labels:
    helm.sh/chart: "GOLDEN_STATIC_VALUE"
    app.kubernetes.io/name: wiz-network-analyzer
    app.kubernetes.io/instance: release-test
    app.kubernetes.io/version: "0.1"
    app.kubernetes.io/managed-by: Helm
  annotations:
    "helm.sh/hook": pre-install, pre-upgrade
    "helm.sh/hook-weight": "-1"
stringData:
  clientId: "client-id"
  clientToken: "client-secret"
